### DISCUSSION TITLE
Is it able to log IP in query log like Pi-hole?

### DISCUSSION THREAD

**Discussion by lessload** [DNSCrypt/dnscrypt-proxy#2107] [Q&A]
I need to lookup ip to domain for diagnos something. but some domain change ip too offen and hard to do that with dnscrypt-proxy

```
16:07:43 dnsmasq[792]: reply 11-courier.push.apple.com is <CNAME>
16:07:43 dnsmasq[792]: reply 11.courier-push-apple.com.akadns.net is <CNAME>
16:07:43 dnsmasq[792]: reply apac-asia-courier-4.push-apple.com.akadns.net is 17.57.145.133
16:07:43 dnsmasq[792]: reply apac-asia-courier-4.push-apple.com.akadns.net is 17.57.145.137
16:07:43 dnsmasq[792]: reply apac-asia-courier-4.push-apple.com.akadns.net is 17.57.145.134
```
Sample log from Pi-hole i can lookup ip `17.57.145.137` is `11-courier.push.apple.com`

Can i do something like this in dnscrypt-proxy ? to make and ip log.
If not. It is a great feature to have it in log option config.

**Comment by lifenjoiner** [comment]
AFAIK, `dnscrypt-proxy` doesn't log the result IP(s).
Some others have an option to show the result including IP(s).
You can:
1. try set one of those servers between `dnscrypt-proxy` and you.
2. try a CLI tool like `dig`, maybe.

**Reply by lessload** [reply]
1. i use it in windows as loopback dns.
2. Windows not have dig or dns log (without MS telemetry) and `dig` doesn't work with some domain that change ip every hour.

- no ip log make it hard to lookup from ip to domain, at the log time.

**Reply by bcookatpcsd** [reply]
Trying to understand the question.. 

dns blocks names.. so dnscrypt-proxy records blocked or passed names.. 

If you needed/wanted to allow something in dnscrypt-proxy.. you wouldn't be using an IP .. 

dnscrypt-proxy does have a -resolve flag.. 

```
# dnscrypt-proxy -config /etc/dnscrypt-proxy/dnscrypt-proxy.toml -resolve dnscrypt.info
Resolving [dnscrypt.info] using 10.20.8.29 port 40

Resolver      : 38.88.8.34 (dns-edge-usa-east-newyork-c.cleanbrowsing.org.)

Canonical name: dnscrypt.info.

IPv4 addresses: 37.59.238.213
IPv6 addresses: -

Name servers  : free.ns.c9x.org., recital.ns.c9x.org.
DNSSEC signed : yes
Mail servers  : 1 mail servers found

HTTPS alias   : -
HTTPS info    : -

Host info     : -
TXT records   : v=spf1 a mx ip4:164.132.13.235 ip4:87.98.134.126 ip4:92.222.49.126 ip4:188.165.248.100 ip4:37.59.238.212/30 ip4:137.74.223.232/30 ip4:82.64.197.71 -all, google-site-verification=FTBiPTbSicuzpH8bxUyL30z3qWwKhnnqPXTZ9qvn0ug

```

Not sure if that gives you something to work with vs what you directly asked for..
